src/_guides/language/concurrency/index.md

```diff
@@ -312,107 +311,144 @@ to implement isolates.
 
  [Flutter `compute()` function]: {{site.flutter-docs}}/cookbook/networking/background-parsing#4-move-this-work-to-a-separate-isolate
 
-
 ### Implementing a simple worker isolate
 
-This section shows the implementation for a
-main isolate and the simple worker isolate that it spawns.
-The worker isolate executes a function and then exits,
-sending the main isolate a single message as it exits.
-(The [Flutter `compute()` function][] works in a similar way.)
+These examples show the implementation of a
+main isolate that spawns a simple worker isolate.
+The examples use [`Isolate.run()`][], 
+which efficiently abstracts the complexity
+of setting up and managing worker isolates manually:
 
-This example uses the following isolate-related API:
+* Spawns (starts and creates) an isolate
+* Takes a function argument and executes
+* Captures the result
+* Returns the result to the main isolate
+* Terminates the isolate once work is complete
+* Checks, captures, and throws exceptions and errors back to the main isolate
 
-* [`Isolate.spawn()`][] and [`Isolate.exit()`][]
-* [`ReceivePort`][] and [`SendPort`][]
+[`Isolate.run()`]: {{site.dart-api}}/dev/dart-isolate/Isolate/run.html
 
-[`Isolate.exit()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/exit.html
-[`Isolate.spawn()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawn.html
-[`ReceivePort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/ReceivePort-class.html
-[`SendPort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/SendPort-class.html
+#### Running an existing method in a new isolate
 
 Here’s the code for the main isolate:
 
 <?code-excerpt "lib/simple_worker_isolate.dart (main)"?>
 ```dart
 void main() async {
   // Read some data.
-  final jsonData = await _parseInBackground();
+  final jsonData = await Isolate.run(_readAndParseJson);
 
-  // Use that data
+  // Use that data.
   print('Number of JSON keys: ${jsonData.length}');
 }
+```
+
+The spawned isolate executes the following code:
```

Comment:
**question:** Is this a spawned isolate or the main isolate? If it's the main, you may want to update this.

Reply:
This is the spawned one, spawned by the main isolate.